rclc/include/rclc/executor.h

```diff
@@ -0,0 +1,387 @@
+// Copyright 2014 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef RCLC__EXECUTOR_H_
+#define RCLC__EXECUTOR_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdio.h>
+#include <stdarg.h>
+
+#include <rcl/error_handling.h>
+#include <rcutils/logging_macros.h>
+
+#include "rclc/executor_handle.h"
+#include "rclc/types.h"
+
+/*! \file executor.h
+    \brief The RCLC-Executor provides an Executor based on RCL in which all callbacks are
+    processed in a user-defined order.
+*/
+
+/* defines the semantics of data communication
+   RCLCPP_EXECUTOR - same semantics as in the rclcpp Executor ROS2(Eloquent)
+   LET             - logical execution time
+*/
+typedef enum
+{
+  RCLCPP_EXECUTOR,
+  LET
+} rclc_executor_semantics_t;
+
+/// Type defintion for trigger function. With the parameters:
+/// - array of executor_handles
+/// - size of array
+/// - application specific struct used in the trigger function
+typedef bool (* rclc_executor_trigger_t)(rclc_executor_handle_t *, unsigned int, void *);
+
+/// Container for RCLC-Executor
+typedef struct
+{
+  /// Context (to get information if ROS is up-and-running)
+  rcl_context_t * context;
+  /// Container for dynamic array for DDS-handles
+  rclc_executor_handle_t * handles;
+  /// Maximum size of array 'handles'
+  size_t max_handles;
+  /// Index to the next free element in array handles
+  size_t index;
+  /// Container to memory allocator for array handles
+  const rcl_allocator_t * allocator;
+  /// Wait set (is initialized only in the first call of the rclc_executor_spin_some function)
+  rcl_wait_set_t wait_set;
+  /// Statistics objects about total number of subscriptions, timers, clients, services, etc.
+  rclc_executor_handle_counters_t info;
+  /// timeout in nanoseconds for rcl_wait() used in rclc_executor_spin_once(). Default 100ms
+  uint64_t timeout_ns;
+  /// timepoint used for spin_period()
+  rcutils_time_point_value_t invocation_time;
+  /// trigger function, when to process new data
+  rclc_executor_trigger_t trigger_function;
+  /// application specific data structure for trigger function
+  void * trigger_object;
+  /// data communication semantics
+  rclc_executor_semantics_t data_comm_semantics;
+} rclc_executor_t;
+
+/**
+ *  Return a rclc_executor_t struct with pointer members initialized to `NULL`
+ *  and member variables to 0.
+ */
+rclc_executor_t
+rclc_executor_get_zero_initialized_executor(void);
+/**
```

Comment:
```suggestion

/**
```

Reply:
fixed in 54f0bbc54a33cc601764ab1d53fe3119d9e61c8b